src/app/core/services/users.py

```diff
@@ -48,4 +50,313 @@ def create_user(email: str, session: ScopedSession):
+
+	Returns:
+		bool: Represents whether user exists
+	"""
+	return session.query(exists(User).where(User.user_id == user_id)).scalar()
+
+def start_user_reset_password(email: str, session: ScopedSession):
+	"""Sends user email with authentication code for password reset
+
+	Args:
+			user_id (int): User ID to check for
+			session (ScopedSession): SQLAlchemy database session
+
+	"""
+	#TODO: we dont have the user id, user has to enter their email to get the code
+	# if user_exists(user_id, session):
+		# user = _get_user_email(user_id, session)
+
+		#TODO: replace the messege with the authenticator code
+		# emailer.send_email("This is a place holder", "Flourish Authentication Code", email)
+	# else:
+	# 	logging.error(f'User does not exist')
+	# 	raise NotFoundError(f'Could not find user with ID: {user_id}')
+
+	emailer.send_email("This is a place holder", "Flourish Authentication Code", email)
+
+
+
+
+
+########################################
+# INTERNAL USE ONLY
+########################################
+
+def _update_password(user_id: int, new_password: str, session: ScopedSession):
+	"""
+	## [DANGER] INTERNAL USE ONLY
+	Updates a user's password.
+
+	Args:
+			user_id (int): [description]
+			new_password (str): [description]
+			session (ScopedSession): [description]
+
+	Raises:
+			NotFoundError: [description]
+			ConflictError: [description]
+			e: [description]
+	"""
+	# TODO: generate new password hash?
+	new_password_hash = new_password
+	try:
+		session.execute(
+			update(User)
+				.where(User.user_id == user_id)
+				.values(password_hash=new_password_hash)
+		)
+		session.commit()
+	except exc.NoResultFound as e:
+		logging.error('Failed to find user')
+		logging.exception(e)
+		raise NotFoundError(f'Could not find user with ID: {user_id}')
+	except exc.IntegrityError as e:
+		logging.error('Failed to update user password')
+		logging.exception(e)
+		raise ConflictError(e)
+	except exc.DatabaseError as e:
+		logging.error('Failed to update user')
+		logging.exception(e)
+		raise e
+
+def _cleanup_password_reset_code(user_id: int, session: ScopedSession):
+	"""
+	## [DANGER] INTERNAL USE ONLY
+	Cleans up the password_reset_code and password_reset_code_expiration columns
+
+	Args:
+			user_id (int): User ID to cleanup
+			session (ScopedSession): SQLAlchemy database session
+
+	Raises:
+			NotFoundError: User not found
+			Exception: Database error
+	"""
+	try:
+		session.execute(
+			update(User)
+				.where(User.user_id == user_id)
+				.values(password_reset_code=None, password_reset_code_expiration=None)
+		)
+		session.commit()
+	except exc.NoResultFound as e:
+		logging.error('Failed to find user')
+		logging.exception(e)
+		raise NotFoundError(f'Could not find user with ID: {user_id}')
+	except exc.DatabaseError as e:
+		logging.error('Failed to cleanup user password reset code')
+		logging.exception(e)
+		raise e
+	
+def _get_user_email(user_id:int, session: ScopedSession):
```

Comment:
I don't think this function is doing what you think it does. It's called get_user_email, but it returns User or None.